From 9198740acdcb26c3d063c2a68076360b33746e09 Mon Sep 17 00:00:00 2001
From: Felipe Oliveira Carvalho <felipekde@gmail.com>
Date: Thu, 18 May 2023 17:20:53 -0300
Subject: [PATCH 8/9] GH-35538: [C++] Remove unnecessary status.h include from
 protobuf (#35673)

### Rationale for this change

Newer versions of protobuf use `absl::Status` instead of `google::protobuf::util::Status`. The status variables in this file are type-annotated with `auto` and we can count on the header that declares the status-returning function we are calling to have included the header that defines `Status`.

### What changes are included in this PR?

Removing an include of a protobuf header that don't exist in later versions of the library.

### Are these changes tested?

I haven't tried building with protobuf 23, but I can be sure this specifically fixes the problem reported in #35538.

Authored-by: Felipe Oliveira Carvalho <felipekde@gmail.com>
Signed-off-by: Weston Pace <weston.pace@gmail.com>
---
 cpp/src/arrow/engine/substrait/serde.cc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cpp/src/arrow/engine/substrait/serde.cc b/cpp/src/arrow/engine/substrait/serde.cc
index f588aff44..faddab36e 100644
--- a/cpp/src/arrow/engine/substrait/serde.cc
+++ b/cpp/src/arrow/engine/substrait/serde.cc
@@ -24,7 +24,6 @@
 #include <google/protobuf/descriptor.h>
 #include <google/protobuf/io/zero_copy_stream_impl_lite.h>
 #include <google/protobuf/message.h>
-#include <google/protobuf/stubs/status.h>
 #include <google/protobuf/util/json_util.h>
 #include <google/protobuf/util/message_differencer.h>
 #include <google/protobuf/util/type_resolver.h>
